test/integration/host_test.rb

```diff
@@ -533,6 +475,29 @@ def test_show_action_buttons
       host.reload
       assert_equal env1.name, host.environment.name
     end
+
+    test 'enables override of hostgroup defaults' do
+      original_environment = FactoryGirl.create(:environment)
+      original_hostgroup = FactoryGirl.create(:hostgroup,
+                                              :environment => original_environment)
+
+      # Clear host environment, then inherit it from hostgroup
+      @host.environment = nil
+      @host.hostgroup = original_hostgroup
+      @host.send(:assign_hostgroup_attributes, ['environment_id'])
+      @host.save
+
+      overridden_hostgroup = FactoryGirl.
+        create(:hostgroup, :environment => FactoryGirl.create(:environment))
+
+      visit edit_host_path(@host)
```

Comment:
I think I have the expectations figured now:

For new hosts:
Each change of the hostgroup should reset the environment. Reason: the user chooses how to create a host, so if he is choosing a hostgroup, previous changes to environment are irrelevant.

For edited hosts:
Changes to hostgroup should not affect environment, unless the user requested inheritance explicitly. Reason: the host is already alive, each change should be validated by the user.